source/basic.tex

```diff
@@ -6306,122 +6306,17 @@
-the value written by $A$, or
-
-\item
-for some evaluation $X$, $A$ is dependency-ordered before $X$ and
-$X$ carries a dependency to $B$.
-
-\end{itemize}
-\begin{note}
-The relation ``is dependency-ordered before'' is analogous to
-``synchronizes with'', but uses release/consume in place of release/acquire.
-\end{note}
-
-\pnum
-An evaluation $A$ \defn{inter-thread happens before} an evaluation $B$
-if
-\begin{itemize}
-\item
-  $A$ synchronizes with $B$, or
-\item
-  $A$ is dependency-ordered before $B$, or
-\item
-  for some evaluation $X$
-  \begin{itemize}
-  \item
-    $A$ synchronizes with $X$ and $X$
-    is sequenced before $B$, or
-  \item
-    $A$ is sequenced before $X$ and $X$
-    inter-thread happens before $B$, or
-  \item
-    $A$ inter-thread happens before $X$ and $X$
-    inter-thread happens before $B$.
-  \end{itemize}
-\end{itemize}
-\begin{note}
-The ``inter-thread happens before'' relation describes arbitrary
-concatenations of ``sequenced before'', ``synchronizes with'' and
-``dependency-ordered before'' relationships, with two exceptions. The first
-exception is that a concatenation never ends with
-``dependency-ordered before'' followed by ``sequenced before''. The reason for
-this limitation is that a consume operation participating in a
-``dependency-ordered before'' relationship provides ordering only with respect
-to operations to which this consume operation actually carries a dependency. The
-reason that this limitation applies only to the end of such a concatenation is
-that any subsequent release operation will provide the required ordering for a
-prior consume operation. The second exception is that a concatenation never
-consist entirely of ``sequenced before''. The reasons for this
-limitation are (1) to permit ``inter-thread happens before'' to be transitively
-closed and (2) the ``happens before'' relation, defined below, provides for
-relationships consisting entirely of ``sequenced before''.
-\end{note}
-
 \pnum
 An evaluation $A$ \defn{happens before} an evaluation $B$
-(or, equivalently, $B$ \defn{happens after} $A$) if
-\begin{itemize}
-\item $A$ is sequenced before $B$, or
-\item $A$ inter-thread happens before $B$.
-\end{itemize}
-The implementation shall ensure that no program execution demonstrates a cycle
-in the ``happens before'' relation.
-\begin{note}
-This cycle would otherwise be
-possible only through the use of consume operations.
-\end{note}
-
-\pnum
-An evaluation $A$ \defn{simply happens before} an evaluation $B$
+(or, equivalently, $B$ happens after $A$)
 if either
 \begin{itemize}
 \item $A$ is sequenced before $B$, or
 \item $A$ synchronizes with $B$, or
-\item $A$ simply happens before $X$ and
-$X$ simply happens before $B$.
+\item $A$ happens before $X$ and $X$ happens before $B$.
 \end{itemize}
 \begin{note}
-In the absence of consume operations,
-the happens before and simply happens before relations are identical.
+An evaluation cannot happen before itself.
```

Comment:
Any chance we could say "does not happen", to avoid problematic modal verbs?

Reply:
Fixed.

Reply:
Thanks!